storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -343,7 +343,7 @@ public LogConfig(Map<?, ?> props) {
         this(props, Collections.emptySet());
     }
 
-    @SuppressWarnings("deprecation")
+    @SuppressWarnings({"deprecation", "this-escape"})
```

Comment:
self note: this warning occurs due to methods such as getBoolean() from AbstractConfig, which is a public class and we can't add `final` to those methods.